examples/reminders/lib/base/di/app_dependencies.dart

```diff
@@ -106,14 +113,26 @@ class _AppDependenciesState extends State<AppDependencies> {
   /// Use different data source regarding of if it is running in web ot not
   List<Provider> get _dataSources => [
         Provider<RemindersFirebaseDataSource>(
-          create: (context) => RemindersFirebaseDataSource(),
+          create: (context) => RemindersFirebaseDataSource(
+            context.read<FlutterSecureStorage>(),
+          ),
+        ),
+        Provider<RealmRemindersDataSource>(
+          create: (context) => RealmRemindersDataSource.getInstance(
+            context.read<FlutterSecureStorage>(),
+            widget.config,
+            const Uuid(),
+          ),
         ),
       ];
 
   List<Provider> get _repositories => [
         Provider<RemindersRepository>(
           create: (context) => RemindersRepository(
-            dataSource: RemindersFirebaseDataSource(),
+            dataSource: (widget.config.environment == EnvironmentType.cloud ||
+                    widget.config.environment == EnvironmentType.local)
+                ? context.read<RealmRemindersDataSource>()
+                : context.read<RemindersFirebaseDataSource>(),
```

Comment:
You can conditionally add the dependencies (data sources), since you are also conditionally initialising the RemindersRepository. This way, we won't have any `RemindersFirebaseDataSource`, for example, for local or cloud environment type.

Reply:
Right, this can be done to prevent initialization of realm when the selected environment is not local or cloud, but `RemindersFirebaseDataSource` also handles anonymous/facebook login